### DISCUSSION TITLE
Groups assignment with OpenID Connect / OAuth2

### DISCUSSION THREAD

**Discussion by drehelis** [requarks/wiki#3773] [Help / Questions] [answered]
I was toying around today with OIDC authentication module and AWS Cognito service.
It's fairly easy to setup and make it work and I'll be documenting and sharing the process.

Setting `Allow self-registration` does create the user based on my Cognito `User Pool` but ignores its group membership. Ideally I'd want the user to be mapped automatically to the corresponding group on `wiki.js` (if exist) in addition to statically auto. assignment, is something like that possible?
```
OS: Docker Container (Linux) - Amazon Linux 2
Wiki.js version: 2.5.197
Database engine: PostgreSQL 11.11
```

**Comment by drehelis** [comment, marked as answer]
Oh, found this right after posting the question above, https://github.com/Requarks/wiki/issues/1874.
I guess it answers my question :-)

**Comment by schewara** [comment]
Had the same Question just now, but as there are no more issues existing in this repo, and the link to the Answer just gives me a 404, it would be great to have the Answer here as well.

**Comment by singularsyntax** [comment]
Hello @drehelis - did you ever document the set up process for AWS Cognito somewhere? It seems like it should be fairly straightforward, but I've been unable to get it to work. And unfortunately, `wiki.js` logging configuration does not seem to work, which does not help matters.

**Reply by drehelis** [reply]
Unfortunately, no. Lacking the requirement at the time made me drop it all together.
